ateam_ai/src/trajectory_generation/trapezoidal_motion_profile.cpp

```diff
@@ -0,0 +1,360 @@
+        sample.accel(i) = trajectory.samples.at(t_idx).accel;
+        is_more_left = true;
+      } else {
+        sample.pose(i) = end(i);
+        sample.vel(i) = end_vel(i);
+        sample.accel(i) = 0;
+      }
+    }
+
+    if (is_more_left) {
+      output.samples.push_back(sample);
+    }
+
+    t += dt;
+    t_idx++;
+  }
+
+  return output;
+}
+
+/**
+ * Returns time (s) to accelerate from start vel to end vel given an acceleration limit
+ */
+double time_to_accelerate(const double start_vel, const double end_vel, const double accel_limit)
+{
+  return std::abs(end_vel - start_vel) / accel_limit;
+}
+
+/**
+ * Return average velocity given the start vel and end vel, assuming a constant acceleration
+ */
+double avg_vel_over_acceleration(const double start_vel, const double end_vel)
+{
+  return std::midpoint(start_vel, end_vel);
+}
+
+/**
+ * Returns distance (m) required to accelerate from start_vel to end_vel given an acceleration limit
+ */
+double dist_covered_during_acceleration(
+  const double start_vel, const double end_vel,
+  const double accel_limit)
+{
+  const double t = time_to_accelerate(start_vel, end_vel, accel_limit);
+  const double avg_vel = avg_vel_over_acceleration(start_vel, end_vel);
+
+  return t * avg_vel;
+}
+
+struct AccelerationPhaseInputs
+{
+  double start_vel;  // m/s
+  double end_vel;  // m/s
+  double accel_limit;  // abs(limiting_accel_value) == m/s^2
+};
+
+struct AccelerationPhaseResults
+{
+  double time_required;
+  double distance_required;
+};
+
+AccelerationPhaseResults get_acceleration_phase_results(const AccelerationPhaseInputs inputs)
+{
+  AccelerationPhaseResults output;
+  output.time_required = time_to_accelerate(inputs.start_vel, inputs.end_vel, inputs.accel_limit);
+  output.distance_required = dist_covered_during_acceleration(
+    inputs.start_vel, inputs.end_vel,
+    inputs.accel_limit);
+
+  return output;
+}
+
+struct PhaseCoeffs
+{
+  double start_time;
+  double start_dist;
+  double start_vel;
+  double duration;
+  double end_vel;
+};
+
+struct TrapezoidalCoeffs
+{
+  //    __________
+  //   /          \.
+  //  /            \.
+  // /              \.
+  // ^^^
+  // Phase 1
+  //    ^^^^^^^^^^
+  //    Phase 2
+  //              ^^^
+  //              Phase 3
+  //
+  // Phase 1 - Accel to some "max" velocity
+  // Phase 2 - Stay at some "max" velocity (optional)
+  // Phase 3 - Decell to the end velocity
+  PhaseCoeffs phase_1;
+  PhaseCoeffs phase_2;
+  PhaseCoeffs phase_3;
+};
+
+Trajectory1d sample_trapezoidal_profile(
+  const TrapezoidalCoeffs coeffs, const double accel_limit,
+  const double dt)
+{
+  const auto sq = [](const double x) {return x * x;};
+  const auto position = [sq](const double t, const double v_initial, const double a) {
+      return v_initial * t + 1. / 2. * a * sq(t);
+    };
+  const auto velocity = [sq](const double t, const double v_initial, const double a) {
+      return v_initial + a * t;
+    };
```

Comment:
Captures, but doesn't use `sq`
```suggestion
  const auto velocity = [](const double t, const double v_initial, const double a) {
      return v_initial + a * t;
    };
```